### DISCUSSION TITLE
How to get path Downloads in File Provider Storage on ios

### DISCUSSION THREAD

**Discussion by weedcry** [dotnet/maui#15293] [Q&A]
I can only save the file to the data of a third-party app.
```c#
var fileManager = NSFileManager.DefaultManager;
var documentDirectory = fileManager.GetUrls(NSSearchPathDirectory.DocumentDirectory, NSSearchPathDomain.All)[0];
``` 
![Capture](https://github.com/dotnet/maui/assets/44254503/f185e9b0-5893-4821-ab0f-1953d8751a47)

 I want to save the file to the Downloads folder of iOS but I can't figure out how to get its path.

**Comment by xtuzy** [comment]
Others app support it? i feel ios not support it. i google it, don't get information about it. xamarin.iOS doc have some doc about save file https://learn.microsoft.com/en-us/xamarin/ios/app-fundamentals/file-system

**Reply by weedcry** [reply]
> Others app support it? i feel ios not support it. i google it, don't get information about it. xamarin.iOS doc have some doc about save file https://learn.microsoft.com/en-us/xamarin/ios/app-fundamentals/file-system

Hi @xtuzy, thank for your help. I have read that. There is also similar documentation on Maui, but it only gets the path of third-party applications. I don't see any document mentioning how to get this Downloads folder.

**Reply by xtuzy** [reply]
Do you have flutter's link about it? i google it , don't get what i want. If you can search how to do it use swift or object-c, you can use new bing or chatgpt to convert it to c#

**Reply by Xyncgas** [reply]
Try Xamarin.Essentials, ChatGPT saying the toolkit contains stuffs you can use for accessing out of the application's internal storage which is usually disallowed by default on mobile devices

**Comment by King4Eli** [comment]
SpecialFolder
